rj_gameplay/rj_gameplay/tactic/line_tactic.py

```diff
@@ -1,30 +1,51 @@
-import stp
-from rj_gameplay.role import dumb_move
 from typing import List, Tuple
 
+import numpy as np
+import stp
 from rj_msgs.msg import RobotIntent
 
+from rj_gameplay.role import dumb_move
+
 
 class LineTactic(stp.tactic.Tactic):
-    """Tactic for line up play that puts all six robots in a line on the left of the field."""
+    """Tactic for line up play that puts all six robots in a line on the left of the field.
+    Assuming that the liners are more than 2"""
 
-    def __init__(self, world_state: stp.rc.WorldState):
+    def __init__(
+        self,
+        world_state: stp.rc.WorldState,
+        num_liners: int,
+        start: np.ndarray,
+        end: np.ndarray,
+    ):
         super().__init__(world_state)
 
         # compute move points
-        # TODO: make start on side of the field, so this Tactic is actually useful during penalty situations
-        start = (3.0, 0.0)
-        dy = 0.5
-        # TODO: make the # here a param instead of hardcoding for same reason as above TODO
-        self.move_points = [(start[0], start[1] + i * dy) for i in range(6)]
+        self.start = start
+        self.end = end
+        xpts = np.linspace(self.start[0], self.end[0], num_liners)
+        ypts = np.linspace(self.start[1], self.end[1], num_liners)
+        self.target_points = []
 
-        # request closest robot every pt
-        for pt in self.move_points:
-            # for some reason stp.role doesn't need to be imported?
+        for x, y in zip(xpts, ypts):
+            target = np.array([x, y])
+            self.target_points.append(target)
             self._role_requests.append(
-                (stp.role.cost.PickClosestToPoint(pt), dumb_move.DumbMove)
+                (stp.role.cost.PickClosestToPoint(target), dumb_move.DumbMove)
```

Comment:
since this tactic is only useful for penalty situations, you could more simply pass in a list of robot ids from the penalty play and force those robots to move into the line. right now if I move the ball around, sometimes the robot that was supposed to be the striker moves into the line.

Reply:
I agree but PickRobotbyId doesn't work due to role assignment failure almost all the time for some reason.
I tick all tactics, so it should work properly now.